### DISCUSSION TITLE
User Warning: Trying to infer the 'batch_size' from an ambigous collection.

### DISCUSSION THREAD

**Discussion by fibr1900** [open-edge-platform/anomalib#1527] [Q&A] [answered]
Hi,

I am trying to run reverse_distillation on the MVTec dataset, but I get this error and then my process is killed:

![MicrosoftTeams-image](https://github.com/openvinotoolkit/anomalib/assets/60730319/c224ef5b-2c6f-4ff5-ae8a-563d6dc5dae1)

I am using the default config settings. I had the same problem with CFlow and GANomaly. Other models I've tried (PaDiM, PatchCore, CFA, STFPM, and DFKDE) I had no problems with.

I am not sure how to fix this error myself. Does anyone know why this is happening? Thankful for help!

**Comment by willyfh** [comment, marked as answer]
Hi @fibr1900 , one possible reason is because the GPU memory is not big enough to train the models. For reverse distillation, you can try to use a smaller backbone, by changing the default backbone from `wide_resnet50_2` to `resnet18`.

**Comment by blaz-r** [comment]
As @willyfh already suggested, it might be a GPU memory issue. The warning regarding the batch size you are getting is just about the logging function and shouldn't crash the process.
